src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,36 +19,70 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the person with the given phone number or identified by the index number used in the "
+            + "displayed person list.\n"
+            + "Parameters: p/PHONE_NUMBER (must be a valid phone number) or i/INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " p/81234567, " + COMMAND_WORD + " i/1";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
+    private final Phone targetPhone;
     private final Index targetIndex;
+    private final boolean isPhonePrefix;
 
+    /**
+     * @param targetPhone of the person in the person list to delete
+     */
+    public DeleteCommand(Phone targetPhone) {
+        this.targetPhone = targetPhone;
+        this.targetIndex = null;
+        this.isPhonePrefix = true;
+    }
+
+    /**
+     * @param targetIndex of the person in the filtered person list to delete
+     */
     public DeleteCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
+        this.targetPhone = null;
+        this.isPhonePrefix = false;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        if (isPhonePrefix) {
+            List<Person> list = model.getAddressBook().getPersonList();
+            int matches = (int) list.stream().filter(p -> p.getPhone().equals(targetPhone)).count();
+
+            if (matches == 0) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_PHONE_NUMBER);
+            }
+
+            Person personToDelete = list.stream().filter(p -> p.getPhone().equals(targetPhone))
```

Comment:
this filtering is repeated here, would it be better to do collect the filtered list first, before doing the checks?